test/PerpetualRouterFactory.t.sol

```diff
@@ -4,30 +4,157 @@ pragma solidity ^0.8.0;
 import {Test} from "forge-std/Test.sol";
 
 import {PerpetualRouterFactory} from "src/PerpetualRouterFactory.sol";
+import {PerpetualPositionRouter} from "src/PerpetualPositionRouter.sol";
 import {PerpetualContracts} from "test/PerpetualContracts.sol";
 
+interface IBadRouterFactory {
+  enum BadRouterType {
+    PositionRouterType,
+    DepositRouterType,
+    MadeUpRouter
+  }
+
+  function deploy(BadRouterType type_, address asset) external returns (address);
+
+  function computeAddress(BadRouterType type_, address asset) external returns (address);
+}
+
 contract RouterFactoryTest is Test, PerpetualContracts {
-  function test_deployPositionRouter() public {
+  PerpetualRouterFactory factory;
+
+  event RouterDeployed(PerpetualRouterFactory.RouterType indexed type_, address indexed asset);
+}
+
+contract Constructor is RouterFactoryTest {
+  function testFork_CorrectlySetsAllConstructorArgs() public {
+    PerpetualRouterFactory factory = new PerpetualRouterFactory(
+  	clearingHouse,
+  	accountBalance,
+  	vault
+        );
+    assertEq(
+      address(factory.PERPETUAL_CLEARING_HOUSE()),
+      address(clearingHouse),
+      "PERPETUAL_CLEARING_HOUSE was not set correctly"
+    );
+    assertEq(
+      address(factory.PERPETUAL_ACCOUNT_BALANCE()),
+      address(accountBalance),
+      "PERPETUAL_ACCOUNT_BALANCE was not set correctly"
+    );
+    assertEq(
+      address(factory.PERPETUAL_VAULT()), address(vault), "PERPETUAL_VAULT was not set correctly"
+    );
+  }
+}
+
+contract Deploy is RouterFactoryTest {
+  function setUp() public {
+    factory = new PerpetualRouterFactory(
+  	clearingHouse,
+  	accountBalance,
+  	vault
+        );
+  }
+
+  function test_EmitPositionRouterDeployedEvent() public {
+    vm.expectEmit();
+    emit RouterDeployed(PerpetualRouterFactory.RouterType.PositionRouterType, VETH);
+
+    factory.deploy(PerpetualRouterFactory.RouterType.PositionRouterType, VETH);
+  }
+
+  function test_EmitDepositRouterDeployedEvent() public {
+    vm.expectEmit();
+    emit RouterDeployed(PerpetualRouterFactory.RouterType.DepositRouterType, VETH);
+
+    factory.deploy(PerpetualRouterFactory.RouterType.DepositRouterType, VETH);
+  }
+
+  function test_ReturnsPositionRouterAddress() public {
     PerpetualRouterFactory factory =
       new PerpetualRouterFactory(clearingHouse, accountBalance, vault);
```

Comment:
How come in some of these tests we create a new factory instance, but not others? Seems we could use the existing `factory` var here and in the other tests

Reply:
https://github.com/ScopeLift/optimizoors-perpetual-protocol/pull/19/commits/2c64edb67c91d06c28928541756a17d0edc0a800